source/common/grpc/grpc_web_filter.cc

```diff
@@ -3,18 +3,51 @@
 #include <arpa/inet.h>
 
 #include "common/common/base64.h"
+#include "common/common/utility.h"
+#include "common/grpc/common.h"
 #include "common/http/headers.h"
 
+#include "spdlog/spdlog.h"
+
 namespace Envoy {
 namespace Grpc {
 
 // Bit mask denotes a trailers frame of gRPC-Web.
 const uint8_t GrpcWebFilter::GRPC_WEB_TRAILER = 0b10000000;
 
+// Supported gRPC-Web content-types.
+const std::vector<std::string>& GrpcWebFilter::GRPC_WEB_CONTENT_TYPES() const {
+  static const std::vector<std::string> types{
+      Http::Headers::get().ContentTypeValues.GrpcWeb,
+      Http::Headers::get().ContentTypeValues.GrpcWebProto,
+      Http::Headers::get().ContentTypeValues.GrpcWebText,
+      Http::Headers::get().ContentTypeValues.GrpcWebTextProto};
+  return types;
+}
+
+bool GrpcWebFilter::isGrpcWebRequest(const Http::HeaderMap& headers) {
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  if (content_type != nullptr) {
+    for (const std::string& valid : GrpcWebFilter::GRPC_WEB_CONTENT_TYPES()) {
+      if (valid == content_type->value().c_str()) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
 // Implements StreamDecoderFilter.
-// TODO(fengli): Implements the subtypes of gRPC-Web content-type, like +proto, etc.
+// TODO(fengli): Implements the subtypes of gRPC-Web content-type other than proto, like +json, etc.
 Http::FilterHeadersStatus GrpcWebFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
   const Http::HeaderEntry* content_type = headers.ContentType();
+  if (!isGrpcWebRequest(headers)) {
+    // TODO(fengli): Return error response.
+    return Http::FilterHeadersStatus::StopIteration;
```

Comment:
This is a functional change that probably should be in a distinct PR, as it's not strictly stats related.